### PR TITLE
PE188-64 Custom layout for product list page

### PR DESCRIPTION
**Pull request by FelipeFabio-Acid**
## Link to Issue(s) in JIRA:
- [https://linets.atlassian.net/browse/PE188-34](https://linets.atlassian.net/browse/PE188-34)

## Description

- Add new product card component
- Add new product list component
- Override spree filter view to match figma
- Override spree product index view to match figma design

Captura de pantalla 2024-03-08 a la(s) 04 55 54

- Native spree pagination using Kaminari with bootstrap 4
Captura de pantalla 2024-03-08 a la(s) 04 58 24

## Checklist

Before you move on, make sure that:

- [ ] No unintended changes are included
- [ ] Spelling is correct
- [ ] There are tests covering new/changed functionality
- [ ] Rubocop style is passing, and Rspec tests are passing.
- [ ] Documentation has been written (Docs or code)
- [ ] Commits have meaningful names and changes. _CR remarks_-like commits are squashed.
- [ ] Proper labels assigned. Use `WIP` label to indicate that state

### CODE REVIEW

File: app/components/products/card_component.html.erb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    <% cache @cache_key do %>
+      <%= @product_image %>
+      <hr>
+
+      <div class="flex flex-col gap-2">
+        <div class="text-xl text-gray-950 font-semibold" title="<%= @product_name %>">
+          <%= @product_name %>
+        </div>
+        <div class="text-base font-normal text-gray-400">
+          <%= @product_description.truncate(100, separation: ' ') %>
```
Comment (LeonardoSantis-AcidLabs) [inline app/components/products/card_component.html.erb:12]: Esta linea me está dando un error:

image

Probablemente sea bueno añadir un safe operator

File: spec/components/previews/products/list_component_preview.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class Products::ListComponentPreview < ViewComponent::Preview
```
Comment (LeonardoSantis-AcidLabs) [inline spec/components/previews/products/list_component_preview.rb:3]: estoy obteniendo este error al consultar el lookbook del list component preview:

image

que podrá ser?

File: app/views/spree/products/_no_results_with_filters.html.erb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<%= render 'spree/products/filter_and_sort_mobile_buttons' %>
+<%#= render 'spree/products/sort_desktop', permitted_params: permitted_params %>
```
Comment (LeonardoSantis-AcidLabs) [inline app/views/spree/products/_no_results_with_filters.html.erb:2]: Quizas sea bueno eliminar este codigo comentado

File: app/views/spree/products/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    <% if params[:keywords] %>
+      <div class="container d-md-flex pt-md-5">
+        <div class="plp-results-text text-uppercase mx-md-auto py-md-5">
+          <%= "#{@products.total_count} #{Spree.t('plp.results_for')} '#{params[:keywords].upcase}'" %>
+        </div>
+      </div>
+    <% end %>
+
+    <%= render 'spree/products/filter_and_sort_mobile_buttons' %>
+    <%#= render 'spree/products/sort_desktop', permitted_params: permitted_params %>
```
Comment (LeonardoSantis-AcidLabs) [inline app/views/spree/products/index.html.erb:21]: Quizas sea bueno eliminar este codigo comentado

File: spec/components/previews/products/list_component_preview.rb (hunk 2 above)
Comment (FelipeFabio-Acid) [reply spec/components/previews/products/list_component_preview.rb:3]: listo, resolví esto, sin usar el helper que estaba usando para las imagenes. Con una image_tag directo mejor.

Comment (LeonardoSantis-AcidLabs) [reply spec/components/previews/products/list_component_preview.rb:3]: https://github.com/Departamento-TI/cenabast-tienda/assets/156705333/059c3f84-752d-43c4-8ed9-67cbdcfdd931

Se renderiza, pero no se ve realmente como una grilla de productos

La vista de `/products` tambien se esta viendo algo extraña 🤔

https://github.com/Departamento-TI/cenabast-tienda/assets/156705333/d7b65fd3-d2ed-4f1d-ad52-7adefdc775d4

**Pull request merged**